docs/remote-clusters.asciidoc

```diff
@@ -0,0 +1,127 @@
+
+When using remote cluster connections with ECK, the necessary setup depends on where the remote cluster is deployed.
+
+[id="{p}-remote-clusters-connect-internal"]
+==== Connect from an Elasticsearch cluster running in the same Kubernetes cluster
+
+TBD
+
+[id="{p}-remote-clusters-connect-external"]
+==== Connect from an Elasticsearch cluster running outside the Kubernetes cluster
+
+NOTE: While it is technically possible to configure remote cluster connections using older versions of Elasticsearch, this guide only covers the setup for Elasticsearch 7.6 and later. The setup process is significantly simplified in Elasticsearch 7.6 due to improved support for the indirection introduced by Kubernetes services.
+
+You can configure a remote cluster connection to an ECK-managed Elasticsearch cluster from another cluster running outside the Kubernetes cluster as follows:
+
+. Ensure that both clusters trust each other's certificate authority.
+. Configure the remote cluster connection via the Elasticsearch REST API.
+
+For illustration purposes, consider the following example:
+
+* `cluster_one` resides inside Kubernetes and is managed by ECK
+* `cluster_two` is not hosted inside the same Kubernetes cluster as `cluster_one` and may not even be managed by ECK 
+
+To configure `cluster_one` as a remote cluster in `cluster_two`:
+
+
+===== Ensure both clusters trust each others certificate authority
+
+The certificate authority (CA) used by ECK to issue certificates for the Elasticsearch transport layer is stored in a secret named `<cluster_name>-es-transport-certs-public`. Extract the certificate for `cluster_one` as follows:
+
+[source,sh]
+----
+kubectl get secret cluster_one-es-transport-certs-public \
+-o go-template='{{index .data "ca.crt" | base64decode}}' > remote.ca.crt
+----
+
+You then need to configure the CA as one of the trusted CAs in `cluster_two`. If that cluster is hosted outside of Kubernetes, simply add the CA certificate extracted in the above step to the list of CAs in link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#_pem_encoded_files_3[`xpack.security.transport.ssl.certificate_authorities`]
+
+If `cluster_two` is also managed by an ECK instance, proceed as follows:
+
+Create a secret with the CA certificate you just extracted:
+[source,sh]
+----
+kubectl create secret generic remote-certs --from-file=remote.ca.crt
+----
+
+Use this secret to configure `cluster_one`'s CA as a trusted CA in `cluster_two`:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: elasticsearch.k8s.elastic.co/{eck_crd_version}
+kind: Elasticsearch
+metadata:
+  name: cluster_two
+spec:
+  nodeSets:
+  - config:
+      xpack.security.transport.ssl.certificate_authorities:
+      - /usr/share/elasticsearch/config/other/remote.ca.crt
```

Comment:
Double-checking because I have doubts: do we default to appending user-provided config array items with our own managed array items?

Reply:
Yes we changed this explicitly for this use case to merging behaviour